src/styles/themes/material.scss

```diff
@@ -27,3 +27,29 @@ $type-scale: utils.type-scale();
         $type-scale: $type-scale
     );
 }
+
+igc-switch {
+    --size: #{utils.rem(36px)};
+    --thumb: #{utils.rem(20px)};
+    --thumb-offset: #{utils.rem(-1px)};
+}
+
+@keyframes thumb-on {
+    from {
+        margin-inline-start: var(--thumb-offset);
+    }
+
+    to {
+        margin-inline-start: calc(var(--size) - var(--thumb) - var(--thumb-offset));
+    }
+}
+
+@keyframes thumb-off {
+    from {
+        margin-inline-start: calc(var(--size) - var(--thumb) - var(--thumb-offset));
+    }
+
+    to {
+        margin-inline-start: var(--thumb-offset);
+    }
+}
```

Comment:
Let's move this back to the original material theme. Since all animation properties are now controlled by CSS variables it shouldn't be an issue. It will also allow us to keep everything DRY.